src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -258,6 +259,60 @@ def __len__(self) -> int:
         """
         return len(self._buffer)
 
+    def window(
+        self,
+        start: int | datetime,
+        end: int | datetime,
+        fill_gaps: Optional[float] = None,
+        force_copy: bool = False,
+    ) -> ArrayLike:
+        """
+        Access a sub window of the `MovingWindow`.
+
+        Args:
+            start: The start index or datetime of the sub window.
+            end: The end index or datetime of the sub window.
+            fill_gaps: The value to fill gaps with.
+            force_copy: If True, a copy of the underlying buffer is returned.
+
+        Raises:
+            IndexError: if the end index is smaller than the start index.
+
+        Returns:
+            A sub window of the `MovingWindow`.
+        """
+        if len(self._buffer) == 0:
+            raise IndexError("The buffer is empty.")
+        if isinstance(start, datetime) and isinstance(end, datetime):
+            if start > end:
+                raise IndexError(
+                    f"end parameter {end} has to predate start parameter {start}"
+                )
+
+            start_index = self._buffer.datetime_to_index(start)
+            end_index = self._buffer.datetime_to_index(end)
+        else:
+            start_index = start  # type: ignore
```

Comment:
Please use another `isinstance(..)` check here for int, not `type: ignore`. Ignore should only be used as the very last resort.

Reply:
agreed. We should only avoid runtime type checks in hot paths. Also you want to assure that the object to proceed with is an integer.

Reply:
Don't we need to wrap here too? I think we do have to. Might be a bit tricky w.r.t. supporting negative end indices.

Reply:
Matthias if I understand correctly what you mean, there are checks for the wrapping later on

Reply:
Sorry, my comment might have be a bit misleading.

The point is, that we want to pass an index as if that `MovingWindow` would be a regular array. But it isn't, since in fact it's a ringbuffer. Therefore, what we have to do is to map the passed index to the index of the actual ringbuffer, which doesn't happen here.